packages/api/core/src/util/plugin-interface.ts

```diff
@@ -61,15 +61,15 @@ export default class PluginInterface implements IForgePluginInterface {
     });
 
     for (const plugin of this.plugins) {
-      plugin.init(dir, forgeConfig);
+      void plugin.then((plugin) => plugin.init(dir, forgeConfig));
```

Comment:
This is dodge and can go if we follow the above comment

Reply:
@MarshallOfSound Maybe it would be better to actually replace the original promises at this part in the array (and go with traditional `for` loop), but still keep it all as an array? This way, if `plugin.init` throws an error, it is further handled (I guess that's the only part I skipped handling).
I really liked the elegance of `for await` in my solution. And I don't see much reason at `await`-ing promises during the class creation, I think it's best to leave that when we actually need to depend on the promise result. IMO this over-complicates the code without much benefits (other than ending with the rejected promise immediately).